File: ComplaintsApp/app/src/main/java/com/complaints/app/data/model/Complaint.kt
package com.complaints.app.data.model

import com.google.gson.annotations.SerializedName

/**
 * Complaint model — matches the backend complaint JSON from MongoDB.
 */
data class Complaint(
    @SerializedName("_id")                val id: String,
    @SerializedName("title")             val title: String,
    @SerializedName("description")       val description: String,
    @SerializedName("category")          val category: String,
    @SerializedName("submittedBy")       val submittedBy: SubmittedBy?,
    @SerializedName("severity")          val severity: String,          // High | Medium | Low | Unclassified
    @SerializedName("severityConfidence")val severityConfidence: Double?,
    @SerializedName("status")            val status: String,            // pending | in-progress | resolved | rejected
    @SerializedName("adminNotes")        val adminNotes: String?,
    @SerializedName("createdAt")         val createdAt: String
)

data class SubmittedBy(
    @SerializedName("_id")   val id: String?,
    @SerializedName("name")  val name: String?,
    @SerializedName("email") val email: String?
)

File: complaints_system-main/backend/models/Complaint.js
/**
 * models/Complaint.js
 *
 * Defines the shape of a complaint document in MongoDB.
 * Each complaint is linked to the user who submitted it.
 */

const mongoose = require('mongoose');

const complaintSchema = new mongoose.Schema(
  {
    title: {
      type:     String,
      required: [true, 'Title is required'],
      trim:     true,
      minlength: [5,  'Title must be at least 5 characters'],
      maxlength: [100, 'Title cannot exceed 100 characters']
    },
    description: {
      type:     String,
      required: [true, 'Description is required'],
      minlength: [20, 'Description must be at least 20 characters']
    },
    category: {
      type:     String,
      required: true,
      enum: [
        'IT Support',
        'Hostels',
        'Academics',
        'Fees / Finance',
        'Maintenance',
        'Transport',
        'Security / Discipline',
        'Administration',
        'Other'
      ]
    },
    status: {
      type:    String,
      enum:    ['pending', 'in-progress', 'resolved', 'rejected'],
      default: 'pending'
    },

    // ── ML-predicted fields ────────────────────────────────
    severity: {
      type:    String,
      enum:    ['Low', 'Medium', 'High', 'Unclassified'],
      default: 'Unclassified'
    },
    severityConfidence: {
      type:    Number,
      default: null
    },

    // ── Relationships ──────────────────────────────────────
    submittedBy: {
      type:     mongoose.Schema.Types.ObjectId,
      ref:      'User',          // reference to the User collection
      required: true
    },
    adminNotes: {
      type:    String,
      default: ''
    }
  },
  { timestamps: true }
);

module.exports = mongoose.model('Complaint', complaintSchema);
